### PR TITLE
test: basic tests exercising rw

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: tests/lean/run/rewrite.lean [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+-- These are often solved by the implicit `rfl` at the end of `rw`.
+example : f 2 = f 3 := by
+  rw [s]
+
+-- We can rewrite at multiple locations,
+-- specializing the arguments differently at each location.
+-- This behaviour was accidentally broken by
+-- https://github.com/leanprover/lean4/pull/2317
+-- and requiring a revert.
```
Comment (tydeu) [inline tests/lean/run/rewrite.lean:43]: Stylistically, should probably be a `/- ... -/` comment. It's a bit to long for line comments.

**Comment by tydeu** [comment]
Also, the commit should probably have the `test:` prefix instead of `chore:`. Otherwise, LGTM, but I am not a  expert on `rw`.

File: tests/lean/run/rewrite.lean [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+example : [f 1, f 2, f 1] = [f 0, f 0, f 0] := by
+  rw [t]
+  rw [t 2]
+
+-- Rewriting by lemmas with undetermined arguments creates additional goals.
+example : f 2 = f 3 := by
+  rewrite [s]
+  rfl
+
+-- These are often solved by the implicit `rfl` at the end of `rw`.
```
Comment (Kha) [inline tests/lean/run/rewrite.lean:35]: Potential additional test case: the `rfl` should only unfold `reducible`

**Review by Kha — COMMENTED** [review]
Potential additional test cases:
* sequence of rewrites
* empty rewrite (i.e. cheap rfl)
* rewriting by `Iff`
* rewriting touches main goal only
* rewriting with config
* should we have negative test cases like "motive is not type-correct"? Not sure.

**Comment by semorrison** [comment]
I've added these (but not one for "motive not type-correct").

I would like to expose the `occs` parameter through the config in a separate PR, and I will add more tests along with that.

**Review by Kha — APPROVED** [review]

**Pull request merged**